src/test/codewhisperer/util/crossFileContextUtil.test.ts

```diff
@@ -0,0 +1,115 @@
+            const actual = getFileDistance(targetFile, candidateFile, '/')
+            assert.strictEqual(actual, 3)
+        })
+
+        it('distance 4', function () {
+            candidateFile = 'ui/popup/CodeWhispererPopupManager.java'
+            const actual = getFileDistance(targetFile, candidateFile, '/')
+            assert.strictEqual(actual, 4)
+        })
+
+        it('distance 5', function () {
+            candidateFile = 'ui/popup/components/CodeWhispererPopup.java'
+            const actual = getFileDistance(targetFile, candidateFile, '/')
+            assert.strictEqual(actual, 5)
+        })
+
+        it('distance 6', function () {
+            candidateFile = 'ui/popup/components/actions/AcceptRecommendationAction.java'
+            const actual = getFileDistance(targetFile, candidateFile, '/')
+            assert.strictEqual(actual, 6)
+        })
+    })
+
+    describe('getRelevantFiles', function () {
+        let tempFolder: string
+
+        beforeEach(async function () {
+            tempFolder = await makeTemporaryToolkitFolder()
+        })
+
+        afterEach(async function () {
+            await fs.remove(tempFolder)
+        })
+
+        it('should return opened files in the current window and sorted ascendingly by file distance', async function () {
+            const targetFile = path.join(tempFolder, 'service', 'microService', 'CodeWhispererFileContextProvider.java')
+            const fileWithDistance3 = path.join(tempFolder, 'service', 'CodewhispererRecommendationService.java')
+            const fileWithDistance5 = path.join(tempFolder, 'util', 'CodeWhispererConstants.java')
+            const fileWithDistance6 = path.join(tempFolder, 'ui', 'popup', 'CodeWhispererPopupManager.java')
+            const fileWithDistance7 = path.join(tempFolder, 'ui', 'popup', 'components', 'CodeWhispererPopup.java')
+            const fileWithDistance8 = path.join(
+                tempFolder,
+                'ui',
+                'popup',
+                'components',
+                'actions',
+                'AcceptRecommendationAction.java'
+            )
+
+            const filePaths = [
+                targetFile,
+                fileWithDistance8,
+                fileWithDistance5,
+                fileWithDistance7,
+                fileWithDistance3,
+                fileWithDistance6,
+            ]
+            const shuffledFilePaths = shuffleList(filePaths)
+
+            for (const file of shuffledFilePaths) {
+                toFile('', file)
+                const textDocument = await vscode.workspace.openTextDocument(vscode.Uri.file(file))
+                await vscode.window.showTextDocument(textDocument, { preview: false })
+            }
+
+            // to make the target file editor active
+            const textDocument = await vscode.workspace.openTextDocument(vscode.Uri.file(targetFile))
+            const editor = await vscode.window.showTextDocument(textDocument)
```

Comment:
the open/show pair is common, might be worth re-using `openATextEditorWithText` : https://github.com/aws/aws-toolkit-vscode/blob/d66f115ee4e1fed8d87fa6a59bd83a1c5925ac05/src/testInteg/stepFunctions/visualizeStateMachine.test.ts#L74

Reply:
yea make sense